src/nupic/types/Sdr.cpp

```diff
@@ -263,6 +263,15 @@ namespace sdr {
     }
 
 
+    SparseDistributedRepresentation& SparseDistributedRepresentation::operator=(const SparseDistributedRepresentation& value) {
+        if( dimensions.empty() ) {
```

Comment:
maybe the behavior should be no-fail, ie always set dimensions to that of the source/value here?

Reply:
I think that the SDR should never be allowed to change its size.  There are a lot of other pieces of code which assume this (all of the metrics) which would break.  IMO changing the size of an SDR halfway through a program is probably an error.